digitaltwins/samples/device/javascript/pnpSampleDevice.js

```diff
@@ -0,0 +1,194 @@
+  case commandNameBlink: {
+    await sendCommandResponse(request, response, 200, 'blink response');
+    break;
+  }
+  case commandNameTurnOn: {
+    await sendCommandResponse(request, response, 200, 'turn On response');
+    break;
+  }
+  case commandNameTurnOff: {
+    await sendCommandResponse(request, response, 200, 'turn Off response');
+    break;
+  }
+  default: // Is there no way to register for an unknown method?
+    await sendCommandResponse(request, response, 404, 'unknown method');
+    break;
+  }
+};
+
+// TODO Not sure if this needs to be async or not coz we probably dont care if the response rached Hub.
+const sendCommandResponse = async (request, response, status, payload) => {
+  helperLogCommandRequest(request);
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
+    console.log(request.payload);
+  };
+};
+
+const helperCreateReportedPropertiesPatch = (propertiesToReport, componentName, actualResponse) => {
+  const componentPart = componentPrefix + componentName;
+  const patch = { [componentPart]: {} };
+
+  Object.keys(propertiesToReport).forEach((propertyName) => {
+    const propertyContent = { value: propertiesToReport[propertyName] };
+    if (actualResponse) {
+      propertyContent.ac = actualResponse.code;
+      propertyContent.ad = actualResponse.description;
+      propertyContent.av = actualResponse.version;
+    }
+    patch[componentPart][propertyName] = propertyContent;
+  });
+  console.log('The following properties will be updated for component:' + componentName);
+  console.log(patch);
+  return patch;
+};
+
+const updateComponentReportedProperties = (deviceTwin, patch, componentName) => {
+  deviceTwin.properties.reported.update(patch, function (err) {
+    if (err) throw err;
+    console.log('Properties have been reported for component:' + componentName);
+  });
+};
+
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin, componentName) => {
+  const prefixAndComponentName = componentPrefix + componentName;
+  const versionProperty = '$version';
+
+  deviceTwin.on('properties.desired.' + prefixAndComponentName, (delta) => {
+    Object.entries(delta).forEach(([propertyName, propertyValue]) => {
+      propertyUpdateHandler(deviceTwin, componentName, propertyName, null, propertyValue.value, deviceTwin.properties.desired[versionProperty]);
+    });
+  });
+};
+
+
+const sendTelemetryAtInterval = async (deviceClient, componentName, secs) => {
+  let index = 0;
+  const interval = parseFloat(secs)*1000;
+  setInterval( async () => {
```

Comment:
>async () [](start = 15, length = 8)

This is what I think we're not supposed to do.  I don't believe that setInterval is built to handle a promise as its first argument.  The async means that this () => "returns" a promise.

Reply:
i have seen some examples of interval and timeout accepting async callbacks but from our class i believe we should not mix callbacks with async. have removed async/await and this will only run in the `check` phase of event loop now.

Reply:
I thought I looked for examples of giving an async to setInterval or setTimer and didn't find one.  I will check again

---
In reply to: [442562459](https://github.com/Azure/azure-iot-sdk-node/pull/821#discussion_r442562459) [](ancestors = 442562459)